airbyte-ci/connectors/pipelines/pipelines/dagger/actions/system/docker.py

```diff
@@ -8,62 +8,145 @@
 
 from dagger import Client, Container, File, Secret
 from pipelines import consts
-from pipelines.airbyte_ci.connectors.context import ConnectorContext, PipelineContext
-from pipelines.consts import DOCKER_HOST_NAME, DOCKER_HOST_PORT, DOCKER_TMP_VOLUME_NAME, DOCKER_VAR_LIB_VOLUME_NAME
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+from pipelines.consts import (
+    DOCKER_HOST_NAME,
+    DOCKER_HOST_PORT,
+    DOCKER_REGISTRY_MIRROR_URL,
+    DOCKER_TMP_VOLUME_NAME,
+    DOCKER_VAR_LIB_VOLUME_NAME,
+    STORAGE_DRIVER,
+    TAILSCALE_AUTH_KEY,
+)
 from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.contexts.pipeline_context import PipelineContext
 
 
-def with_global_dockerd_service(
-    dagger_client: Client, docker_hub_username_secret: Optional[Secret] = None, docker_hub_password_secret: Optional[Secret] = None
-) -> Container:
-    """Create a container with a docker daemon running.
-    We expose its 2375 port to use it as a docker host for docker-in-docker use cases.
+def get_base_dockerd_container(dagger_client: Client, docker_var_lib_caching: bool = True) -> Container:
```

Comment:
Why do we still have this argument? Also the docstring text doesn't match, regarding the default value. Just remove it, it's always going to be always-on or always-off at all callsites.

Reply:
I'l;l just change this back to always on, like you said